### PR TITLE
FEAT: PostHog Custom Events

### PR DESCRIPTION
**Pull request by tahierhussain**
## What
Added the required posthog custom events.

-

## Why
The default PostHog events that were being triggered were too noisy. Therefore, to avoid this issue, we disabled the default capture and created our custom events, adding them only where necessary.

-

## How
Posthog provides us to create custom events. [Read more](https://posthog.com/docs/libraries/react).

-

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

Yes. This PR has the potential to break existing features in the FE. Prior to this PR, in some instances, the onClick value was directly modified within the JSX part. For example: onClick={() => setData()}. However, including the posthog function call directly within the JSX part wouldn't be ideal. To address this, I've created a new function in such cases and refactored the code accordingly. But I have tested such cases thoroughly to make sure any existing features don't break.

-

## Database Migrations
NA
- 

## Env Config
NA
- 

## Relevant Docs
https://posthog.com/docs/libraries/react
-

## Related Issues or PRs
Cloud PR: https://github.com/Zipstack/unstract-cloud/pull/82
-

## Dependencies Versions
NA
-

## Notes on Testing
NA
-

## Screenshots
NA

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

**Review by Deepak-Kesavan — APPROVED** [review]

File: frontend/src/components/helpers/auth/RequireAdmin.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ const RequireAdmin = () => {
   if (!isAdmin) {
     return <Unauthorized />;
   }
-  if (isOpenSource) {
+  if (!isOpenSource) {
```
Comment (Deepak-Kesavan) [inline frontend/src/components/helpers/auth/RequireAdmin.js:16]: @tahierhussain Why was this changed?

File: frontend/src/components/input-output/configure-ds/ConfigureDs.jsx [hunk 2]
```diff
@@ -151,7 +168,7 @@ function ConfigureDs({
   };
 
   const handleSubmit = () => {
-    if (!isTcSuccessful) {
+    if (isTcSuccessful) {
```
Comment (Deepak-Kesavan) [inline frontend/src/components/input-output/configure-ds/ConfigureDs.jsx:171]: @tahierhussain If test connection is successful, the current logic will throw an error.

Comment (tahierhussain) [reply frontend/src/components/input-output/configure-ds/ConfigureDs.jsx]: @Deepak-Kesavan Thanks for the catch. I had made this change for testing purpose. I have reverted the necessary changes.

File: frontend/src/components/helpers/auth/RequireAdmin.js (hunk 1 above)
Comment (tahierhussain) [reply frontend/src/components/helpers/auth/RequireAdmin.js]: @Deepak-Kesavan Thanks for the catch. I had made this change for testing purpose. I have reverted the necessary changes.

**Review by nehabagdia — APPROVED** [review]
